streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java

```diff
@@ -118,49 +134,68 @@ public void intermediateSuppressionShouldThrow() {
     }
 
 
-    @Test
-    public void finalResultsSuppressionShouldThrow() {
-        final KTableSuppressProcessor<Windowed<String>, Long> processor =
-            new KTableSuppressProcessor<>(finalResults(ofMillis(1)));
+    @Test(expected = KTableSuppressProcessor.NotImplementedException.class)
+    public void finalResultsSuppressionShouldBufferAndEmitLater() {
+        final KTableSuppressProcessor<Windowed<String>, Long> processor = new KTableSuppressProcessor<>(
+            finalResults(ofMillis(1L)),
+            timeWindowedSerdeFrom(String.class),
+            new FullChangeSerde<>(Long())
+        );
 
         final MockInternalProcessorContext context = new MockInternalProcessorContext();
         processor.init(context);
 
-        context.setTimestamp(ARBITRARY_LONG);
-        try {
-            processor.process(new Windowed<>("hey", ARBITRARY_WINDOW), new Change<>(ARBITRARY_LONG, ARBITRARY_LONG));
-            fail("expected an exception for now");
-        } catch (final KTableSuppressProcessor.NotImplementedException e) {
-            // expected
-        }
+        final long timestamp = ARBITRARY_TIMESTAMP;
+        context.setRecordMetadata("topic", 0, 0, null, timestamp);
+        final Windowed<String> key = new Windowed<>("hey", ARBITRARY_WINDOW);
+        final Change<Long> value = ARBITRARY_CHANGE;
+        processor.process(key, value);
         assertThat(context.forwarded(), hasSize(0));
+
+        assertThat(context.scheduledPunctuators(), hasSize(1));
+        context.scheduledPunctuators().get(0).getPunctuator().punctuate(timestamp + 1L);
+
+        assertThat(context.forwarded(), hasSize(1));
+        final MockProcessorContext.CapturedForward capturedForward = context.forwarded().get(0);
+        assertThat(capturedForward.keyValue(), is(new KeyValue<>(key, value)));
+        assertThat(capturedForward.timestamp(), is(timestamp));
     }
 
-    @Test
-    public void finalResultsWith0GraceBeforeWindowEndShouldThrow() {
-        final KTableSuppressProcessor<Windowed<String>, Long> processor =
-            new KTableSuppressProcessor<>(finalResults(ofMillis(0)));
+    @Test(expected = KTableSuppressProcessor.NotImplementedException.class)
+    public void finalResultsWith0GraceBeforeWindowEndShouldBufferAndEmitLater() {
```

Comment:
nit: I realize this was here in a previous commit, but in this test and the next spell out `Zero` vs using the number, it's hard to read.